timesketch/frontend/src/components/SketchExploreSessionChart.vue

```diff
@@ -0,0 +1,308 @@
+                    </select>
+                </div>
+            </div>
+        </div>
+   </div>
+</template>
+
+<script>
+import TsVegaLiteChart from './VegaLiteChart'
+import ApiClient from '../utils/RestApiClient'
+
+export default {
+  name: 'ts-explore-session-chart',
+  components: {
+        TsVegaLiteChart
+  },
+
+  data () {
+      return {
+            processedSessions: [],
+            message: 'Show Session Chart',
+            showChart: false,
+            showTimeRange: false,
+            selectedSessions: [],
+            spec: '{}',
+            sessionTypes: ['all'],
+            barSize: 0,
+            smallestTimestamp: 0,
+            selectedType: '',
+            selectedID: '',
+            sessions: [],
+            startTimeRange: Date.now() - 31556952000,
+            endTimeRange: Date.now(),
+            timeRangeMessage: '',
+            isTruncated: false
+      }
+  },
+
+  computed: {
+    sketch () {
+        return this.$store.state.sketch
+    },
+    indices () {
+        return this.$store.state.currentQueryFilter.indices
+    }
+  },
+
+  methods: {
+    registerClickListener: function (view) {
+        var store = this.$store
+        var sketchId = this.sketch.id
+        view.addEventListener('click', function (event, item) {
+            var session_type = item.datum.session_type
+            var session_id = item.datum.session_id
+            if (session_type != undefined && session_id != undefined) {
+                var queryString = session_type + ':' + session_id
+                store.commit('updateCurrentQueryString', queryString)
+                store.commit('search', sketchId)
+            }
+        })
+    },
+
+    toggleChart: function () {
+        this.showChart = !this.showChart
+        if (this.showChart) {
+            this.message = 'Hide Session Chart'
+            if (this.sessions.length == 0) {
+                ApiClient.getSessions(this.sketch.id, this.indices[0]).then((response) => {
+                    this.sessions = response.data
+                    this.sessionTypes = this.sessionTypes.concat([...new Set(this.sessions.map(s => s.session_type))])
+                    this.getVegaSpec()
+                }).catch((e) => {})
+            }
+            else {
+                this.getVegaSpec()
+            }
+        }
+        else {
+            this.message = 'Show Session Chart'
+            this.selectedSessions = []
+            this.spec = '{}'
+            this.selectedType = ''
+            this.selectedID = ''
+            this.showTimeRange = false
+            this.startTimeRange  = Date.now() - 31556952000,
+            this.endTimeRange = Date.now(),
+            this.timeRangeMessage = ''
+        }
+    },
+
+    toggleTimeRange: function () {
+        this.showTimeRange = !this.showTimeRange
+    },
+    
+    updateTimeRange: function (event) {
+        if(/\d{1,2}\/\d{1,2}\/\d{4}/.test(event.target.value)) {
```

Comment:
Same here, can you put the RE in a constant with a descriptive name to make this a bit more readable?